### ISSUE TITLE
Error: Type 'P_2' cannot be used to index type 'FilledImageFieldImage & Record<ThumbnailNames extends string ? Exclude<ThumbnailNames, "url" | "dimensions" | "alt" | "copyright"> : never, FilledImageFieldImage>'

### ISSUE THREAD

**Issue by jeremytenjo** [prismicio/prismic-helpers#55]
### Versions

```json
    "@prismicio/client": "6.6.3",
    "@prismicio/helpers": "2.3.2",
    "@prismicio/next": "0.1.3",
    "@prismicio/react": "2.4.4",
```

- node: v16.15.0

### Steps to reproduce

1. 
```sh
git clone --branch prismic-preview-fix https://github.com/jeremytenjo/starter-website.git && cd starter-website && npm i && npm run tests:ts-types
```

### What is expected?

Terminal to have no errors after running npm run tests:ts-types

### What is actually happening?

Getting error after running npm run tests:ts-types:
 
```sh
node_modules/@prismicio/helpers/dist/index.d.ts:309:1463 - error TS2536: Type 'P_2' cannot be used to index type 'FilledImageFieldImage & Record<ThumbnailNames extends string ? Exclude<ThumbnailNames, "url" | "dimensions" | "alt" | "copyright"> : never, FilledImageFieldImage>'.

309 }, ThumbnailNames extends Exclude<keyof Field, number | symbol | "url" | "dimensions" | "alt" | "copyright">>(field: Field | ((_prismicio_types.EmptyImageFieldImage & Record<ThumbnailNames extends string ? Exclude<ThumbnailNames, "url" | "dimensions" | "alt" | "copyright"> : never, _prismicio_types.EmptyImageFieldImage | _prismicio_types.FilledImageFieldImage> extends infer T ? { [P in keyof T]: (_prismicio_types.EmptyImageFieldImage & Record<ThumbnailNames extends string ? Exclude<ThumbnailNames, "url" | "dimensions" | "alt" | "copyright"> : never, _prismicio_types.EmptyImageFieldImage | _prismicio_types.FilledImageFieldImage>)[P]; } : never) | (_prismicio_types.FilledImageFieldImage & Record<ThumbnailNames extends string ? Exclude<ThumbnailNames, "url" | "dimensions" | "alt" | "copyright"> : never, _prismicio_types.EmptyImageFieldImage | _prismicio_types.FilledImageFieldImage> extends infer T_1 ? { [P_1 in keyof T_1]: (_prismicio_types.FilledImageFieldImage & Record<ThumbnailNames extends string ? Exclude<ThumbnailNames, "url" | "dimensions" | "alt" | "copyright"> : never, _prismicio_types.EmptyImageFieldImage | _prismicio_types.FilledImageFieldImage>)[P_1]; } : never)) | null | undefined) => field is _prismicio_types.FilledImageFieldImage & Record<ThumbnailNames extends string ? Exclude<ThumbnailNames, "url" | "dimensions" | "alt" | "copyright"> : never, _prismicio_types.FilledImageFieldImage> extends infer T_2 ? { [P_2 in keyof T_2]: (_prismicio_types.FilledImageFieldImage & Record<ThumbnailNames extends string ? Exclude<ThumbnailNames, "url" | "dimensions" | "alt" | "copyright"> : never, _prismicio_types.FilledImageFieldImage>)[P_2]; } : never;
```

**Comment by angeloashmore** [comment]
Hi @jeremytenjo, thanks for reporting this issue. We are working on fixing the underlying issue here: https://github.com/prismicio/prismic-types/pull/46

I'll keep this issue open so we can track the `@prismicio/types` dependency issue in `@prismicio/helpers`.

**Comment by jeremytenjo** [comment]
Great to know! Thanks @angeloashmore

**Comment by angeloashmore** [comment]
@jeremytenjo This issue should be fixed now as of `v2.3.3`. Could you give it a try and let me know if the issue is fixed for you?

You can upgrade to the latest version of `@prismicio/helpers` with the following command:

```sh
npm install @prismicio/helpers@latest
```

I'm going to close the issue since it should be fixed, but let me know if it doesn't and I'll reopen it and reinvestigate. Thanks again! 🙂

**Comment by jeremytenjo** [comment]
Works great, thank you @angeloashmore !!
